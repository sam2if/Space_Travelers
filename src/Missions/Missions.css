.missions {
  padding: 25px;
}

table {
  width: 100%;
  border-right: 1px solid black;
  border-top: 1px solid black;
  border-bottom: 1px solid black;
}

tr:nth-child(odd) {
  background-color: #d6eeee;
}

th {
  background-color: white !important;
  text-align: center;
  padding: 20px;
  border-left: 1px solid black;
}

td {
  text-align: center;
  padding: 20px;
  border-left: 1px solid black;
}
