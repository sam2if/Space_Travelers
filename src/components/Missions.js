import React from 'react';

function Missions() {
  return (
    <div>
      <h2>Missions Page</h2>
      {/* Add the actual content for Missions page here */}
    </div>
  );
}

export default Missions;
