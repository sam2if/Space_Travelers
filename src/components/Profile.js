import React from 'react';

function Profile() {
  return (
    <div>
      <h2>My Profile Page</h2>
      {/* Add the actual content for My Profile page here */}
    </div>
  );
}

export default Profile;
