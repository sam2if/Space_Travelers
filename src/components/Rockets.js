import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import css from './Rockets.module.css';
import { fetchRockets, setReserve } from '../redux/rockets/rocketsSlice';
import LoadingSpinner from './LoadingR';

function Rockets() {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rockets.allRockets);
  const { loading } = useSelector((state) => state.rockets);

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  if (loading) {
    return <LoadingSpinner />;
  }

  return (
    <>
      {rockets.map((rocket) => (
        <div key={rocket.id} className={css['rocket-container']}>
          <div className={css['rocket-image-container']}>
            <img src={rocket.flickr_images} alt="rocket" />
          </div>
          <div className={css['rocket-info']}>
            <p className={css['rocket-title']}>{rocket.name}</p>
            <p className={css['rocket-desc']}>
              {rocket.reserved && <span className={css['reserved-badge']}>Reserved</span>}
              {rocket.description}
            </p>

            <button
              type="button"
              className={rocket.reserved ? css['cancel-reservation-button'] : css['reserve-button']}
              onClick={() => {
                dispatch(setReserve({ id: rocket.id, reserved: !rocket.reserved }));
              }}
            >
              {rocket.reserved ? 'Cancel Reservation' : 'Reserve Rocket'}
            </button>
          </div>
        </div>
      ))}
    </>
  );
}

export default Rockets;
