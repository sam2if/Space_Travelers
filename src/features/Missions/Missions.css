:root {
  --border: 1px solid #5e6062;
}

.missions {
  padding: 25px;
  display: flex;
  justify-content: center;
}

.mission-name {
  font-weight: bold;
}

table {
  width: 90%;
  border-right: var(--border);
  border-top: var(--border);
}

tr:nth-child(odd) {
  background-color: #f1f2f2;
}

th {
  background-color: white !important;
  text-align: center;
  padding: 20px;
  border-left: var(--border);
  border-bottom: var(--border);
}

td {
  padding: 20px;
  border-left: var(--border);
  border-bottom: var(--border);
}

.mstatus {
  width: max-content;
}

td button {
  width: max-content;
  padding: 8px;
  border-radius: 5px;
}

.mstat {
  border-radius: 5px;
  padding: 5px;
  width: max-content;
  color: white;
  font-weight: bold;
}

.member {
  background-color: rgb(33, 168, 185);
}

.notmember {
  background-color: rgb(70, 72, 72);
}

.leavebtn {
  border: 2px solid red;
  color: red;
}

.joinbtn {
  border: 2px solid rgb(70, 72, 72);
}

@media screen and (max-width: 767px) {
  /* Styles for small screens */
  table {
    width: 100%;
  }

  th,
  td {
    padding: 10px;
    font-size: 14px;
  }
}
