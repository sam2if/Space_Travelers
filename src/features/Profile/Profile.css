.profile {
  display: flex;
  justify-content: space-around;
}

.listtitle {
  margin-bottom: 9px;
  padding: 0;
  font-size: xx-large;
}

.listbox {
  display: flex;
  flex-direction: column;
  text-align: start;
  width: 45%;
  height: max-content;
}

.itemlist {
  all: unset;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  border: 1px solid rgb(101, 101, 103);
}

.listitems {
  border: 1px solid rgb(101, 101, 103);
  list-style: none;
  font-size: larger;
  height: auto;
  padding: 20px;
}

@media screen and (max-width: 767px) {
  /* Styles for small screens */
  .profile {
    flex-direction: column;
    align-items: center;
  }

  .listbox {
    width: 100%;
    margin-bottom: 20px;
  }
}
