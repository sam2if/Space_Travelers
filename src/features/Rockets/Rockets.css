.rockets {
  display: flex;
  flex-direction: column;
}

.rocketimg {
  height: 150px;
  aspect-ratio: 4 / 3;
}

.rocketTile {
  display: flex;
  align-items: center;
  padding: 10px;
  margin-bottom: 8px;
}

.reserved {
  margin-right: 10px;
  font-size: medium;
  background-color: #15b8d4;
  color: white;
  width: fit-content;
  width: -moz-fit-content;
  padding: 6px;
  border-radius: 8px;
  font-weight: bolder;
}

.notReserved {
  display: none;
}

.rocketDetails {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  margin-left: 20px;
}

.rocketName {
  font-size: larger !important;
  margin: 0;
}

.rocketDescription {
  max-width: 800px;
  margin-top: 10px;
  margin-bottom: 10px;
  line-height: 28px;
}

.reserveButton {
  all: unset;
  padding: 8px;
  text-align: center;
  border-radius: 8px;
  color: white;
  width: fit-content;
  width: -moz-fit-content;
}

.reserveButton:hover {
  cursor: pointer;
}

.cancel {
  cursor: pointer;
  background-color: rgb(220, 224, 227);
  color: black;
}

.reserve {
  background-color: #3177f8;
}

@media screen and (max-width: 767px) {
  /* Styles for small screens */
  .rocketTile {
    flex-direction: column;
    align-items: flex-start;
  }

  .rocketimg {
    margin-bottom: 10px;
  }

  .rocketDetails {
    margin-left: 0;
    align-items: flex-start;
  }
}
