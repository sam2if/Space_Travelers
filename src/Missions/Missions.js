import React from 'react';
import './Missions.css';

function Missions() {
  const missions = [
    {
      id: 1,
      mission_name: 'mars',
      mission_description: 'dffffffffffffffffffffffffffff',
      joined: false,
    },
    {
      id: 2,
      mission_name: 'mars',
      mission_description: 'dffffffffffffffffffffffffffff',
      joined: false,
    },
    {
      id: 3,
      mission_name: 'mars',
      mission_description: 'dffffffffffffffffffffffffffff',
      joined: false,
    },
  ];
  return (
    <div className="missions">
      <table cellPadding="0" cellSpacing="0">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {
            missions.map((elem) => (
              <tr key={elem.id}>
                <td>{elem.mission_name}</td>
                <td>{elem.mission_description}</td>
                <td>{elem.joined}</td>
                <td><button type="button">Join Mission</button></td>
              </tr>
            ))
          }
        </tbody>

      </table>
    </div>
  );
}

export default Missions;
