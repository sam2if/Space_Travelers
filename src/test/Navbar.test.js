import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Profile from '../features/Profile/Profile';
import Missions from '../features/Missions/Missions';
import store from '../store/store';
import Rockets from '../features/Rockets/Rockets';

it('renders  profile correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders  mission correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders  Rocket correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
