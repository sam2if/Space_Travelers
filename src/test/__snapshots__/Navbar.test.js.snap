// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders  Rocket correctly 1`] = `
<div>
  Loading...
</div>
`;

exports[`renders  mission correctly 1`] = `
<div
  className="missions"
>
  <table
    cellPadding="0"
    cellSpacing="0"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`renders  profile correctly 1`] = `
<div
  className="profile"
>
  <div
    className="listbox"
  >
    <h2
      className="listtitle"
    >
      My Missions
    </h2>
    <ul
      className="itemlist"
    />
  </div>
  <div
    className="listbox"
  >
    <h2
      className="listtitle"
    >
      My Rockets
    </h2>
    <ul
      className="itemlist"
    />
  </div>
</div>
`;
